jans-cedarling/cedarling/src/common/cedar_schema/cedar_json/action.rs

```diff
@@ -5,27 +5,180 @@
+    pub schema_entities: &'a CedarSchemaEntities,
+    pub schema: &'a ActionSchema,
+}
+
+impl Action<'_> {
+    /// Builds the JSON representation of context entities for a given action.
+    ///
+    /// This method processes the context attributes of the action and generates a
+    /// corresponding JSON value. The context may include entity references (with
+    /// `type` and `id`) and other values, which can be mapped through the provided
+    /// `id_mapping` and `value_mapping`.
+    ///
+    /// The `id_mapping` param is a A `HashMap` that maps context attribute keys
+    /// (like `"access_token"`) to their corresponding `id`s (like `"acs-tkn-1"`).
+    ///
+    /// # Usage Example
+    ///
+    /// ```rs
+    /// let id_mapping = HashMap::from([("access_token".to_string(), "acs-tkn-1".to_string())]);
+    /// let json = action.build_ctx_entities_json(id_mapping, value_mapping);
+    /// ```
+    pub fn build_ctx_entity_refs_json(
+        &self,
+        mut id_mapping: HashMap<String, String>,
+    ) -> Result<Value, BuildJsonCtxError> {
+        let mut json = json!({});
+
+        for attr in self.context_entities.iter() {
+            if let CedarType::TypeName(type_name) = &attr.kind {
+                let id = match id_mapping.remove(&attr.key) {
+                    Some(val) => val,
+                    None => Err(BuildJsonCtxError::MissingIdMapping(attr.key.clone()))?,
+                };
+                let type_name = format!("{}::{}", attr.namespace, type_name);
+                json[attr.key.clone()] = json!({"type": type_name, "id": id});
+            }
+        }
+
+        Ok(json)
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum BuildJsonCtxError {
+    /// If an entity reference is provided but the ID is missing from `id_mapping`.
+    #[error("An entity reference for `{0}` is required by the schema but an ID was not provided via the `id_mapping`")]
+    MissingIdMapping(String),
+    /// If a non-entity attribute is provided but the value is missing from `value_mapping`.
+    #[error("A non-entity attribute for `{0}` is required by the schema but a value was not provided via the `value_mapping`")]
+    MissingValueMapping(String),
+}
+
+impl CedarSchemaJson {
+    /// Find the action in the schema
+    pub fn find_action(
+        &self,
+        action_name: &str,
+        namespace: &str,
+    ) -> Result<Option<Action>, FindActionError> {
+        let schema_entities = match self.namespace.get(namespace) {
+            Some(entities) => entities,
+            None => return Ok(None),
+        };
+
+        let action_schema = match schema_entities.actions.get(action_name) {
+            Some(schema) => schema,
+            None => return Ok(None),
+        };
+
+        let mut principal_entities = HashSet::new();
+        for res_type in action_schema.principal_types.iter().map(|s| s.as_str()) {
+            principal_entities.insert(format!("{}::{}", namespace, res_type));
```

Comment:
In codebase exist `CEDAR_POLICY_SEPARATOR` constant, maybe it would be better to move it to better place and use it here?

Reply:
sure, changed here: [54836b5](https://github.com/JanssenProject/jans/pull/10387/commits/54836b5ec63298894fc3f9189dad96f98898285c)